### DISCUSSION TITLE
<ItemGroup>add .cs files and embedded resources

### DISCUSSION THREAD

**Discussion by khoshbin** [3F/MvsSln#49] [Q/A]
I want to add some c# files and folders to .csproj file:

    <Compile Include="financial\Invoice.cs" />
    <Compile Include="financial\Invoice.Designer.cs">
      <DependentUpon>Invoice.cs</DependentUpon>
    </Compile>

also I want to add some **embedded resource**:

```

  <ItemGroup>
    <EmbeddedResource Include="Images\Images.Invoice.png" />
```

Is it possible with your library? if yes, how?

Regards

**Comment by 3F** [comment]
If something is not available through XProject directly, you can through msbuild since MvsSln 2.x partially relies on it when processing project instances. While in MvsSln 3.x this is planned to be replaced by IeXod.

For MvsSln 2.5.3 try this, for example:

```csharp
using var sln = new Sln("...sln", SlnItems.Env);

sln.Result.Env
    .LoadProjects(sln.Result.ProjectItemsConfigs.Where(p => p.project.IsCs()))
    .ForEach(xp =>
    {
        xp.AddItem("Compile", @"financial\Invoice.cs");
        xp.AddItem("Compile", 
                    @"financial\Invoice.Designer.cs", 
                    new[] { new KeyValuePair<string, string>("DependentUpon", "Invoice.cs") }
        );
        xp.AddItem("EmbeddedResource", @"Images\Images.Invoice.png");
        xp.Save();
    });
```

**Comment by khoshbin** [comment]
thank you so much
